buildpacktoml_test.go

```diff
@@ -286,32 +286,71 @@ func testBuildpackTOML(t *testing.T, when spec.G, it spec.S) {
 					}
 				})
 
-				it("should include processes", func() {
-					h.Mkfile(t,
-						`[[processes]]`+"\n"+
-							`type = "some-type"`+"\n"+
-							`command = "some-cmd"`+"\n"+
+				when("processes", func() {
+					it("should include processes and use the default value that is set", func() {
+						h.Mkfile(t,
 							`[[processes]]`+"\n"+
-							`type = "other-type"`+"\n"+
-							`command = "other-cmd"`+"\n",
-						filepath.Join(appDir, "launch-A-v1.toml"),
-					)
-					br, err := bpTOML.Build(lifecycle.BuildpackPlan{}, config)
-					if err != nil {
-						t.Fatalf("Unexpected error:\n%s\n", err)
-					}
-					if s := cmp.Diff(br, lifecycle.BuildResult{
-						BOM:         nil,
-						Labels:      []lifecycle.Label{},
-						MetRequires: nil,
-						Processes: []launch.Process{
-							{Type: "some-type", Command: "some-cmd", BuildpackID: "A"},
-							{Type: "other-type", Command: "other-cmd", BuildpackID: "A"},
-						},
-						Slices: []layers.Slice{},
-					}); s != "" {
-						t.Fatalf("Unexpected metadata:\n%s\n", s)
-					}
+								`type = "some-type"`+"\n"+
+								`command = "some-cmd"`+"\n"+
+								`default = true`+"\n"+
+								`[[processes]]`+"\n"+
+								`type = "web"`+"\n"+
+								`command = "other-cmd"`+"\n",
+							// default is false and therefore doesn't appear
+							filepath.Join(appDir, "launch-A-v1.toml"),
+						)
+						br, err := bpTOML.Build(lifecycle.BuildpackPlan{}, config)
+						if err != nil {
+							t.Fatalf("Unexpected error:\n%s\n", err)
+						}
+						if s := cmp.Diff(br, lifecycle.BuildResult{
+							BOM:         nil,
+							Labels:      []lifecycle.Label{},
+							MetRequires: nil,
+							Processes: []launch.Process{
+								{Type: "some-type", Command: "some-cmd", BuildpackID: "A", Default: true},
+								{Type: "web", Command: "other-cmd", BuildpackID: "A", Default: false},
+							},
+							Slices: []layers.Slice{},
+						}); s != "" {
+							t.Fatalf("Unexpected metadata:\n%s\n", s)
+						}
+					})
+					when("when there are multiple default=true processes but they are all of the same type", func() {
```

Comment:
Hmm... I don't think this is expected behavior. A buildpack shouldn't be able to declare multiple processes of the same type.

Reply:
You're right that a builpdpack shouldn't be able to declare multiple processes of the same type but do we check it anywhere?
I don't think that we check it in `builpacktoml_test.go` before introducing the new functionality.
What is the expected behavior?
We thought this is a good practice but maybe we just want to fail.

Reply:
I think it was always possible for an individual buildpack to declare multiple processes of the same type, but because we were putting those processes into a map keyed by type, in the end the list only has one process per type. Maybe we should update the spec to be more clear on this point before introducing a new failure... but I think we could remove this test to avoid suggesting that this is something buildpacks _should_ be doing. What do you think?

Reply:
I totally agree with everything you wrote. I'm just wondering about deleting this test.
I think that I prefer to keep it so the behavior will be clear for the reader. I can write a comment that explains that it's not something that buildpacks should do.
But I definitely think that we should make a spec change and then fail whenever this thing happens.
What do you think?

Reply:
Responded here: https://github.com/buildpacks/lifecycle/pull/520#discussion_r568975384

Reply:
I'm marking this conversation as resolved because I think [this](https://github.com/buildpacks/lifecycle/pull/520#discussion_r568685339) conversation is dealing with the same concern..